Stream_support/include/CGAL/IO/read_3mf.h

```diff
@@ -0,0 +1,658 @@
+            NMR::lib3mf_release(pBuildItem);
+            NMR::lib3mf_release(pBuildItemIterator);
+            NMR::lib3mf_release(pModel);
+            std::cerr << "could not get object transform: " << std::hex << hResult << std::endl;
+            return -1;
+          }
+        }
+        else {
+          Transform = transform_nmr_internal::initMatrix();
+        }
+
+        if(func(pMeshObject, Transform, points, triangles, colors, name)){
+          all_points.push_back(points);
+          all_polygons.push_back(triangles);
+          all_colors.push_back(colors);
+          names.push_back(name);
+        }
+      }
+    }
+    // Release Object Resource ID
+    NMR::lib3mf_release(pObjectResource);
+    // Release Build Item
+    NMR::lib3mf_release(pBuildItem);
+
+    // Move to next Item
+    hResult = NMR::lib3mf_builditemiterator_movenext(pBuildItemIterator, &pbHasNext);
+    if (hResult != LIB3MF_OK) {
+      std::cerr << "could not get next build item: " << std::hex << hResult << std::endl;
+      return -1;
+    }
+  }
+  // Release Build Item Iterator
+  NMR::lib3mf_release(pBuildItemIterator);
+  return all_points.size();
+}
+
+/*!
+ * \brief read_soups_from_3mf extracts ranges of points and triangles from the
+ * `MeshObject`s contained in `file_name`
+ * \tparam PointRanges a model of the concepts `RandomAccessContainer` and
+ *  `BackInsertionSequence` whose `value type` is
+ * a model of the concepts `RandomAccessContainer` and `BackInsertionSequence`
+ *  whose `value type` is the point type.
+ * \tparam PolygonRanges a model of the concept `RandomAccessContainer` whose
+ *  `value_type` is a model of the concept `RandomAccessContainer`
+ * whose `value_type` is a model of the concept `RandomAccessContainer` whose
+ *  `value_type` is std::size_t.
+ * \tparam ColorRanges a model of the concepts `RandomAccessContainer` and
+ *  `BackInsertionSequence` whose `value type` is
+ * a model of the concepts `RandomAccessContainer` and `BackInsertionSequence`
+ *  whose `value type` is `CGAL::Color`.
+ * \param file_name the name of the 3mf file to read.
+ * \param all_points a `PointRanges` that will contain the points of the meshes
+ *  in `file_name`.
+ * Each of these meshes will add a range of its points.
+ * \param all_polygons a `PolygonRanges` that will contain the triangles of the
+ *  meshes in `file_name`.
+ * Each of these meshes will add a range of its triangles. A `triangle` of
+ *  all_polygons[i] contains the indices of its points in all_points[i].
+ * \param all_colors will contain the color of each triangle for each soup.
+ * \param names will contain the name of each mesh in `file_name` if any.
+ *  If the i'th mesh has no name, it will be called "Unknown Mesh" in names.
+ * \return the number of meshes processed in `file_name`.
+ */
+template<typename PointRanges, typename PolygonRanges, typename ColorRanges>
+int read_soups_from_3mf(const std::string& file_name, PointRanges& all_points,
```

Comment:
I don't like the name. `read_triangle_soups_from_3mf` is a bit better. It would be nice to have a function `read_3mf` like we have `read_off` that expects the soups to be valid meshes. Maybe that new function could also skip the names, colors, ...